### PR TITLE
docs(v2): Refactor + Articles Infra

### PR DESCRIPTION
**Pull request by eranhirsch**
Adding a heading section to the v1 docs that would provide general context for the v2 migration

### CODE REVIEW

File: docs/src/components/markdown.astro
Comment (cjquines) [inline docs/src/components/markdown.astro:19]: **thought:** oh this makes a lot more sense

**Comment by eranhirsch** [comment]
This is a huge PR, I played around a lot with astro to try and see how we can make it the easiest to add content to the site, and re-use as much as possible now that V1 is going to be part of the site for a while.

The main things in this PR are:
* Adding the ability to render content as part of the docs pages
* Normalizing the usage of shadcn, removing any manual changes done previously in favour of simpler modifications "from the outside".
* Creating a version selector in the navbar instead of 2 entries in the top bar.

File: docs/src/content/docs/v1/minimum-versions.md
Comment (cjquines) [inline docs/src/content/docs/v1/minimum-versions.md:31]: **question:** will these have descriptions eventually? should we TODO this?

File: docs/src/components/function-api.astro
Comment (cjquines) [inline docs/src/components/function-api.astro:23]: **nit:** bold [underline](https://practicaltypography.com/underlining.html) is a bit too much emphasis imo, just the bold is fine

File: docs/tsconfig.json
Comment (cjquines) [inline docs/tsconfig.json:8]: **thought:** not sure how i feel about this...

i think it's nice, but it does obscure the import

if you look at [shadcn itself](https://ui.shadcn.com/docs/components/accordion#usage), it always imports from `@/components/ui`

File: docs/src/pages/index.astro
Comment (cjquines) [inline docs/src/pages/index.astro:1]: **question:** is there some way to get this on markdown? i guess the lead section and the install stuff can stay here but i feel like the codeblocks can be put in markdown, but idk enough about astro

File: docs/src/content/docs/v1/migration-intro.md
Comment (cjquines) [inline docs/src/content/docs/v1/migration-intro.md:1]: **praise:** nice

File: docs/src/content/config.ts
Comment (cjquines) [inline docs/src/content/config.ts:23]: **nit:** maybe this is premature, but i'd rather this be a literal of versions rather than a boolean

File: docs/src/components/version-selector.tsx
Comment (cjquines) [inline docs/src/components/version-selector.tsx:1]: **praise:** thanks; i was gonna do something like this but you got to it first!

File: docs/src/components/tag-badge.tsx
Comment (cjquines) [inline docs/src/components/tag-badge.tsx:14]: **thought:** this is one of the things i dislike about the way we compile tailwind... you'd think we could just do a variable with the color, but that doesn't work!

**Review by cjquines — APPROVED** [review]

File: docs/src/content/docs/v1/minimum-versions.md
Comment (eranhirsch) [reply docs/src/content/docs/v1/minimum-versions.md:31]: They will. I wanted to send out the actual content in a separate PR but didn't want to handle merge conflicts, so I put the content here until you approve this PR.

File: docs/tsconfig.json
Comment (eranhirsch) [reply docs/tsconfig.json:8]: I agree; it was an idea I had to make it more straightforward for junior contributors to understand where these components come from if they want to change or add new ones. I feel that Shadcn does a lousy job documenting itself in a project. I can see how a project deteriorates in quality as people who don't know what Shadcn is start mucking around with the copied components.

File: docs/src/components/tag-badge.tsx
Comment (eranhirsch) [reply docs/src/components/tag-badge.tsx:14]: I don't think these colors are even very good, but because in v2 we only have one tag, I wouldn't put too much effort into optimizing this. The result is cute and makes sense, and this is the easiest way to do it.

File: docs/src/components/version-selector.tsx
Comment (eranhirsch) [reply docs/src/components/version-selector.tsx:1]: I looked around, copied this design specifically from eslint, they have a similar approach of a single "live" version, and a single previous version which is completely deprecated where the docs are only there to aid the transition (they don't maintain docs for v7, v6, etc...).

File: docs/src/content/config.ts
Comment (eranhirsch) [reply docs/src/content/config.ts:23]: I'm thinking of doing this via the folder (`v1/`) instead of having any sort of version flag for the same reason that I've mentioned in other places: I don't want us to "maintain" a v1 site, hopefully, what we're doing now is the last we'll touch anything related to v1. I prefer not to build the whole site in a way that signals multi-version support.

My goal with this abstraction is to add extra documentation articles to the docs so that we remove the explanations from the index page so we can make it splashier and less verbose.

File: docs/src/pages/index.astro
Comment (eranhirsch) [reply docs/src/pages/index.astro:1]: We can move the whole thing to something like MDX; it's probably pretty easy to do. However, I still think we'll change the main page considerably and don't want to put in the effort yet. We can defer to another PR; this PR is long enough as it is.

**Pull request merged**
